packages/world/src/systems/MineSystem.sol

```diff
@@ -341,12 +345,105 @@ library MineLib {
 }
 
 library RandomResourceLib {
+  struct RandomDrop {
+    ObjectType objectType;
+    uint256[] distribution;
+  }
+
   function _getMineDrops(EntityId mined, ObjectType objectType, Vec3 coord) public view returns (ObjectAmount[] memory) {
-    return NatureLib.getMineDrops(mined, objectType, coord);
+    RandomDrop[] memory randomDrops = _getRandomDrops(mined, objectType);
+
+    ObjectAmount[] memory result = new ObjectAmount[](randomDrops.length + 1);
+
+    if (randomDrops.length > 0) {
+      uint256 randomSeed = NatureLib.getRandomSeed(coord);
+      for (uint256 i = 0; i < randomDrops.length; i++) {
+        RandomDrop memory drop = randomDrops[i];
+        (uint256 cap, uint256 remaining) = NatureLib.getCapAndRemaining(drop.objectType);
+        uint256[] memory weights = RandomLib.adjustWeights(drop.distribution, cap, remaining);
+        uint256 amount = RandomLib.selectByWeight(weights, randomSeed);
+        result[i] = ObjectAmount(drop.objectType, uint16(amount));
+      }
+    }
+
+    // If farmland, convert to dirt
+    if (objectType == ObjectTypes.Farmland || objectType == ObjectTypes.WetFarmland) {
+      objectType = ObjectTypes.Dirt;
+    }
+
+    // Add base type as a drop for all objects
+    result[result.length - 1] = ObjectAmount(objectType, 1);
+
+    return result;
+  }
+
+  function _getRandomDrops(EntityId mined, ObjectType objectType) private view returns (RandomDrop[] memory drops) {
```

Comment:
why was this moved back into the system vs kept in the lib?

Reply:
Because it was only used here